src/ronin/StakingVesting.sol

```diff
@@ -63,40 +64,107 @@ contract StakingVesting is
     _setFastFinalityRewardPercentage(fastFinalityRewardPercent);
   }
 
-  function initializeV4(uint256 activatedAtPeriod, uint256 fastFinalityRewardPercentREP10) external reinitializer(4) {
-    _rep10ActivationPeriod = activatedAtPeriod;
-    _fastFinalityRewardPercentageREP10 = fastFinalityRewardPercentREP10;
-    emit FastFinalityRewardPercentageUpdatedForREP10(fastFinalityRewardPercentREP10);
+  function initializeV4(
+    uint256, /* activatedAtPeriod */
+    uint256 /* fastFinalityRewardPercentREP10 */
+  ) external reinitializer(4) { }
+
+  function initializeV5(
+    uint64 minRewardAmount,
+    uint64 maxRewardAmount,
+    Reward[] calldata rewards
+  ) external reinitializer(5) {
+    _setBlockRewardRange(minRewardAmount, maxRewardAmount);
+    _updateBlockRewards(rewards);
   }
 
   /**
    * @inheritdoc IStakingVesting
    */
-  function getREP10ActivatedAtPeriod() external view override returns (uint256) {
-    return _rep10ActivationPeriod;
+  function updateBlockRewards(
+    Reward[] calldata rewards
+  ) external onlyAdmin {
+    _updateBlockRewards(rewards);
   }
 
   /**
    * @inheritdoc IStakingVesting
    */
-  function receiveRON() external payable { }
+  function getBlockRewards() external view returns (Reward[] memory) {
+    return _blockRewards;
+  }
 
   /**
    * @inheritdoc IStakingVesting
    */
-  function blockProducerBlockBonus(
-    uint256 /* _block */
-  ) public view override returns (uint256) {
-    return _blockProducerBonusPerBlock;
+  function setBlockRewardRange(uint64 minRewardAmount, uint64 maxRewardAmount) external onlyAdmin {
+    _setBlockRewardRange(minRewardAmount, maxRewardAmount);
   }
 
   /**
    * @inheritdoc IStakingVesting
    */
-  function bridgeOperatorBlockBonus(
-    uint256 /* _block */
-  ) public view override returns (uint256) {
-    return _bridgeOperatorBonusPerBlock;
+  function getBlockRewardRange() external view returns (uint64 minRewardAmount, uint64 maxRewardAmount) {
+    return (_minRewardAmount, _maxRewardAmount);
+  }
+
+  /**
+   * @dev See {IStakingVesting-setBlockRewardRange}.
+   */
+  function _setBlockRewardRange(uint64 minRewardAmount, uint64 maxRewardAmount) internal {
+    if (minRewardAmount > maxRewardAmount) revert ErrInvalidArguments(msg.sig);
+    _minRewardAmount = minRewardAmount;
+    _maxRewardAmount = maxRewardAmount;
+    emit BlockRewardRangeUpdated(msg.sender, minRewardAmount, maxRewardAmount);
+  }
+
+  /**
+   * @dev See {IStakingVesting-updateBlockRewards}.
+   */
+  function _updateBlockRewards(
+    Reward[] calldata rewards
+  ) internal {
+    uint256 length = rewards.length;
+    if (length == 0) revert ErrEmptyArray();
+    delete _blockRewards;
```

Comment:
This just sets the array length to zero and leaves the values at each index untouched, right? Is that expected?